applications/OptimizationApplication/python_scripts/algorithms/algorithm_steepest_descent.py

```diff
@@ -115,11 +123,13 @@ def Solve(self):
 
                 alpha = self.__line_search_method.ComputeStep()
 
-                self.UpdateControlField(alpha)
+                self.ComputeControlUpdate(alpha)
 
-                self.converged = self.__convergence_criteria.IsConverged()
+                self.Output()
 
-                self.CallOnAllProcesses(["output_processes"], Kratos.OutputProcess.PrintOutput)
+                self.UpdateControl()
```

Comment:
Yes, I see, in old version the control_field was from next iteration (after adding update). Good catch, thnx!